nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ExecuteScript.java

```diff
@@ -0,0 +1,585 @@
+     * is what we used to populate the list. So just search the list of factories until a match is
+     * found, then create and return a script engine.
+     *
+     * @return a Script Engine corresponding to the currently specified name, or null if none is found.
+     */
+    private ScriptEngine getScriptEngine() {
+        //
+        ScriptEngineFactory factory = scriptEngineFactoryMap.get(scriptEngineName);
+        if (factory == null) {
+            return null;
+        }
+        return factory.getScriptEngine();
+    }
+
+    /**
+     * Handles changes to this processor's properties. If changes are made to script- or engine-related properties,
+     * the script will be reloaded.
+     *
+     * @param descriptor of the modified property
+     * @param oldValue   non-null property value (previous)
+     * @param newValue   the new property value or if null indicates the property
+     */
+    @Override
+    public void onPropertyModified(final PropertyDescriptor descriptor, final String oldValue, final String newValue) {
+        final ProcessorLog logger = getLogger();
+
+        if (SCRIPT_FILE.equals(descriptor)
+                || SCRIPT_BODY.equals(descriptor)
+                || MODULES.equals(descriptor)
+                || SCRIPT_ENGINE.equals(descriptor)) {
+            lock.lock();
+            try {
+                // if the script is changing we'll want to reload the instance
+                if (SCRIPT_FILE.equals(descriptor)) {
+                    if (isFile(newValue)) {
+                        scriptNeedsReload.set(true);
+
+                        // we're attempted to load the script so we need to watch for updates
+                        scriptWatcher = new SynchronousFileWatcher(Paths.get(newValue), new LastModifiedMonitor());
+                    } else {
+                        // the doesn't appear to be a file
+                        scriptWatcher = null;
+                    }
+
+                    // always want to record the configured value
+                    scriptPath = newValue;
+                } else if (SCRIPT_BODY.equals(descriptor)) {
+
+                    scriptNeedsReload.set(true);
+                    scriptBody = newValue;
+
+                } else if (MODULES.equals(descriptor)) {
+
+                    // temporarily set new value (will be restored to oldValue if something goes wrong)
+                    modulePath = newValue;
+                    try {
+                        setupEngine();
+
+                        // we only want to reload during a module change if the script is already loaded
+                        if (isFile(scriptPath)) {
+                            scriptNeedsReload.set(true);
+                        }
+                    } catch (Throwable t) {
+                        modulePath = oldValue;
+                        logger.error(t.getLocalizedMessage(), t);
```

Comment:
This log message is likely to be very ambiguous without the context of what was going on at the time. Would recommend logging something like:
logger.error("Unable to load modules from {} due to {}", new Object[] {newValue, t});

Should also probably catch Exception, rather than Throwable?